src/util/string.cpp

```diff
@@ -196,6 +197,93 @@ std::string translatePassword(const std::string &playername,
 	return pwd;
 }
 
+void getSRPVerifier(const std::string &playername,
+	const std::string &password, const char ** salt, int * salt_len,
+	const char ** bytes_v, int * len_v)
+{
+	std::string n_playername = playername;
+	std::transform(n_playername.begin(), n_playername.end(),
+		n_playername.begin(), ::toupper);
+	srp_create_salted_verification_key(SRP_SHA256, SRP_NG_4096,
+		n_playername.c_str(), (const unsigned char *) password.c_str(),
+		password.size(), (const unsigned char **) salt, salt_len,
+		(const unsigned char **) bytes_v, len_v, NULL, NULL);
+}
+
+// Get a db-ready SRP verifier
+// The salt param is only modifyable by this method so that you can free it
+// if it was allocated. You shouldn't use it for other purposes, as you will
+// need the contents of salt_len too.
+inline static std::string getSRPVerifier(const std::string &playername,
+	const std::string &password, const char ** salt, int salt_len)
+{
+	std::string n_playername = playername;
+	std::transform(n_playername.begin(), n_playername.end(),
+		n_playername.begin(), ::toupper);
+	char * bytes_v = NULL;
+	int len_v;
+	getSRPVerifier(playername, password, salt, &salt_len,
+		(const char **) &bytes_v, &len_v);
+	std::string ret_val = encodeSRPVerifier(std::string(bytes_v, len_v),
+		std::string((char *) *salt, salt_len));
+	if (bytes_v)
+		free(bytes_v);
+	return ret_val;
+}
+
+// Get a db-ready SRP verifier
+std::string getSRPVerifier(const std::string &playername,
+	const std::string &password)
+{
+	char * salt = NULL;
+	std::string ret_val = getSRPVerifier(playername,
+		password, (const char **) &salt, 0);
+	if (!salt)
```

Comment:
You don't need to check if salt is not NULL  (i.e. free(NULL); does not cause a problem)